### PR TITLE
Ocular implanter

### PR DESCRIPTION
**Pull request by flappybatpal**
[QOL][FEATURE][ADDTOWIKI]

## About the PR 
Replaces the ocular prothesis (SecHUD) kit with a terrifying fully automated eye implanter. 
Has a ten second action bar and does one eye at a time. Checks for eyewear and hats with COVERSEYES, informing the user they need to be removed. Causes slight bleeding.
Sprites by Gannets.

## Why's this needed? 
The ocular prothesis kit can spawn in surplus crates and as spy bounties but is easily attainable for 2100 credits from DOC. Eye surgery is also risky and borderline suicidal for clowns.
The implanter makes taking advantage of this much simpler and safer.

Start sound used: [DVD drive spinning up, reading, spinning down(ogg)](https://freesound.org/people/Darth_Biomech/sounds/398441/) by [Darth_Biomech](https://freesound.org/people/Darth_Biomech/), used under [CC BY](https://creativecommons.org/licenses/by/3.0/). Changes made: Trimmed to use a short section, compressed and converted into OGG.
End sound used: https://freesound.org/people/joeliomartini/sounds/89603/ used under [CC0](https://creativecommons.org/publicdomain/zero/1.0/)

## Changelog

```changelog
(u)Flappybat & Gannets
(*)Replaced the ocular prothesis (SecHUD) kit from surplus crates with a fully automatic implanter.
```

### CODE REVIEW

**Comment by flappybatpal** [comment]
This has code as if it could do both eyes but I couldn't work out a menu to do it. You can choose left or right because that's how eye surgery currently works.
The sound is pretty long and AFAIK there's no way to cut it. If the action bar is interrupted this could make for a jank experience.

**Review by virvatuli-gh — CHANGES_REQUESTED** [review]
With this long of a sound there really should be a code prevention in place to prevent it from being spammed like an `ON_COOLDOWN`, for at least as long as it takes for the sound to fully play (especially since the sound is deliberately a little abrasive). Optimally the sound should cut off if the action bar is interrupted but unsure about the feasibility of doing this.

**Comment by flappybatpal** [comment]
Changed this to a short start and end sound.

File: code/obj/item/device/ocular_implanter.dm [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+					else
+						user.show_text("This implanter doesn't contain the left eye implant.")
+
+
+	proc/start_replace_eye(var/target, var/mob/living/carbon/human/H)
+		if(H.glasses)
+			boutput(H, "<span class='alert'>You need to remove your eyewear first.</span>")
+			return
+		if (H.head && H.head.c_flags & COVERSEYES)
+			boutput(H, "<span class='alert'>Your headware covers your eyes, you need to remove it first.</span>")
```
Comment (virvatuli-gh) [inline code/obj/item/device/ocular_implanter.dm:40]:
```suggestion
			boutput(H, "<span class='alert'>Your headwear covers your eyes, you need to remove it first.</span>")
```
makes it consistent with "eyewear" above

File: code/obj/item/device/ocular_implanter.dm [hunk 2]
```diff
@@ -0,0 +1,93 @@
+#define EYE_LEFT 1
+#define EYE_RIGHT 2
+#define EYE_BOTH 4
+
+/obj/item/device/ocular_implanter
+	name = "Ocular Implanter (SecHUD)"
+	icon_state = "ocular_implanter-full"
+	desc = "A worrying looking medical device for automated eye implants, this model is for SecHUDs. The suction cup fill you with dread."
```
Comment (virvatuli-gh) [inline code/obj/item/device/ocular_implanter.dm:8]:
```suggestion
	desc = "A worrying looking medical device for automated eye implants, this model is for SecHUDs. The suction cup fills you with dread."
```

**Review by virvatuli-gh — CHANGES_REQUESTED** [review]
Just a few grammatical things. Doing the sounds this way works! The end sound caught me off guard and gave me a giggle. Everything else looks good to me and functions fine!

**Review by virvatuli-gh — APPROVED** [review]

**Pull request merged**